portality/tasks/reporting.py

```diff
@@ -209,6 +193,35 @@ def count(self, prov):
             self._count_down(self._last_period)
             self._last_period = p
 
+
+    def _get_best_role(self, roles):
+        role_precedence = ["associate_editor", "editor", "admin"]
+        best_role = None
+        for r in roles:
+            try:
+                if best_role is None and r in role_precedence:
+                    best_role = r
+                if role_precedence.index(r) > role_precedence.index(best_role):
+                    best_role = r
+            except ValueError:
+                pass                            # The user has a role not in our precedence list (e.g. api) - ignore it.
+
+        return best_role
+
+
+    def _is_countable(self, prov, role):
+        countable = False
+
+        if role == "admin" and (prov.action == "status:accepted" or prov.action == "status:rejected"):
+            countable = True
+        elif role == "editor" and prov.action == "status:ready":
+            countable = True
+        elif role == "associate_editor" and prov.action == "status:completed":
+            countable = True
+
+        return countable
```

Comment:
Could be simplified as

```
if ...
   return True
elif ...
   return True
else:
    return False
```

but that's inconsequential, really.

Reply:
Well, that's a bit of code style I'd say. What you're suggesting does fit with general python recommended style. Makes code easier to read (when applied 100 or 1000 times), so it matters slightly.

There are some style guides which might be useful to consider for the refactor actually - provided they fit us as a team.